libs/core/src/lib/calendar/calendar.component.html

```diff
@@ -8,6 +8,8 @@
     (closeClicked)="closeClicked.emit()"
     (nextClicked)="handleNextArrowClick()"
     (previousClicked)="handlePreviousArrowClick()"
+    [previousButtonDisabled]="previousButtonDisableFunction(selectedDate)"
+    [nextButtonDisabled]="nextButtonDisableFunction(selectedDate)"
```

Comment:
This will be invoked on every CD cycle. Any reasons for this? 
According to my comment above:

1. Checks should be applied whenever bindings are changed, date is changed or date view is changed
2. As a context it would be better to pass selected date and selected view type
3. Arguments should be passed to the function
4. You should also check if provided value is actually a function
5. Extract type of the function to a separate type, that can be used outside of the component